### DISCUSSION TITLE
Feature Request: Choose directory (F1/F2) with cursor keys

### DISCUSSION THREAD

**Discussion by Amadeus6767** [mikekazakov/nimble-commander#162] [Ideas]
Hi,

I hope it is a simple request and I have the feeling it worked before. But now it does not.
If I press "F1" to search for a directory/ harddisk and then start typing - nimble commander finds all the devices in which the phrase appears.

For example. I Have 6 harddisk named : Foto_1, Foto_2, Foto_3, Foto_4, 8TB_1_Foto, 8TB_2_Foto
If I type "8TB", nimble commander shows me the last 2 devices.

**Now the request:** I would like to choose the second device "8TB_2_Foto", JUST by pressing the "Cursor DOWN" key.

That does not work. Would it be possible to implement that feature?

Regards

Amadeus6767

**Comment by mikekazakov** [comment]
Hi, you are right, navigation via arrows worked in these menus but broke in Sonoma due to some internal changes in MacOS itself. I looked into the issue briefly before releasing v1.5 but wasn't able to find a way to fix it cheaply. So need to investigate it further and maybe, in the worst-case scenario, roll out a custom implementation of these filterable lists which will not be based on MacOS menus at all.
Any help is welcome, the source code is here:
https://github.com/mikekazakov/nimble-commander/blob/main/Source/CUI/include/CUI/FilterPopUpMenu.h
https://github.com/mikekazakov/nimble-commander/blob/main/Source/CUI/source/FilterPopUpMenu.mm
https://github.com/mikekazakov/nimble-commander/blob/main/Source/NimbleCommander/NimbleCommander/States/FilePanels/Actions/ShowGoToPopup.h
https://github.com/mikekazakov/nimble-commander/blob/main/Source/NimbleCommander/NimbleCommander/States/FilePanels/Actions/ShowGoToPopup.mm

**Comment by mikekazakov** [comment]
Opened the issue #187, will be fixed in v1.6, closing this discussion.
